testcases/MachineConfig.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+# IBM_PROLOG_BEGIN_TAG
+# This is an automatically generated prolog.
+#
+# $Source: op-test-framework/testcases/MachineConfig.py $
+#
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2024
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+
+import os
+import unittest
+import time
+
+import OpTestConfiguration
+import OpTestLogger
+from common.OpTestSystem import OpSystemState
+
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class MachineConfig(unittest.TestCase):
+
+
+    def setUp(self):
+
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)
+        self.bmc_type = conf.args.bmc_type
+        if self.bmc_type == "FSP_PHYP" or self.bmc_type == "EBMC_PHYP" :
+            self.hmc_user = conf.args.hmc_username
+            self.hmc_password = conf.args.hmc_password
+            self.hmc_ip = conf.args.hmc_ip
+            self.lpar_name = conf.args.lpar_name
+            self.system_name = conf.args.system_name
+            self.cv_HMC = self.cv_SYSTEM.hmc
+            self.lpar_prof = conf.args.lpar_prof
+        else:
+            self.skipTest("Functionality is supported only on LPAR")
+
+
+class LparConfig(MachineConfig):
+
+    '''
+    pass lpar_config in config file indicating proc mode, vtpm and vpmem.
+    valid values: cpu=shared or cpu=dedicated
+                  vtpm=1 or vtpm=0
+                  vpmem=0 or vpmem=1
+    Ex: lpar_config="cpu=dedicated,vtpm=1,vpmem=1"
+    ''' 
+
+    def setUp(self):
+        super(LparConfig, self).setUp()
+        conf = OpTestConfiguration.conf
+        try: self.lpar_config = conf.args.lpar_config
+        except AttributeError:
+            self.lpar_config = "cpu=shared, vtpm=1, vpmem=1"
```

Comment:
can we also enhance this as like if any requested config is enabled in the system we should skip